scanpipe/models.py

```diff
@@ -3777,6 +3780,99 @@ def as_spdx(self):
         )
 
 
+class PackageScore(UUIDPKModel, PackageScoreMixin):
+
+    def __str__(self):
+        return self.score or str(self.uuid)
+
+    discovered_package = models.ForeignKey(
+        DiscoveredPackage,
+        related_name="discovered_packages_score",
+        help_text=_("The package for which the score is given"),
+        on_delete=models.CASCADE,
+        editable=False,
+        blank=True,
+        null=True,
+    )
+
+    @classmethod
+    @transaction.atomic()
+    def create_from_data(cls, DiscoveredPackage, scorecard_data, scoring_tool=None):
+        """Create ScoreCard Object from ScoreCard Object"""
+        final_data = {
+            "score": scorecard_data.score,
+            "scoring_tool_version": scorecard_data.scoring_tool_version,
+            "scoring_tool_documentation_url": (
+                scorecard_data.scoring_tool_documentation_url
+            ),
+        }
+
+        date_str = scorecard_data.score_date
+        if date_str:
+
+            naive_datetime = datetime.strptime(date_str, "%Y-%m-%d")
+
+            score_date = timezone.make_aware(
+                naive_datetime, timezone.get_current_timezone()
+            )
+        else:
+            score_date = timezone.now()
+
+        final_data["score_date"] = score_date
+
+        scorecard_object = cls.objects.create(
+            **final_data,
+            discovered_package=DiscoveredPackage,
+            scoring_tool=scoring_tool,
+        )
+
+        # Create associated scorecard_checks
+        checks_data = scorecard_data.checks
+
+        ScorecardCheck.objects.bulk_create(
+            [
+                ScorecardCheck(
+                    check_name=check_data.check_name,
+                    check_score=check_data.check_score,
+                    reason=check_data.reason or "",
+                    details=check_data.details or [],
+                    for_package_score=scorecard_object,
+                )
+                for check_data in checks_data
+            ]
+        )
+
+        return scorecard_object
+
+
+class ScorecardCheck(UUIDPKModel, ScorecardChecksMixin):
+
+    def __str__(self):
+        return self.check_score or str(self.uuid)
+
+    for_package_score = models.ForeignKey(
+        PackageScore,
+        related_name="discovered_packages_score_checks",
+        help_text=_("The checks for which the score is given"),
+        on_delete=models.CASCADE,
+        editable=False,
+        blank=True,
+        null=True,
+    )
+
+    @classmethod
+    def create_from_data(cls, package_score, check_data):
+        """Create a ScorecardCheck instance from provided data."""
+        final_data = {
+            "check_name": check_data.get("name"),
+            "check_score": check_data.get("score"),
+            "reason": check_data.get("reason"),
+            "details": check_data.get("details", []),
+            "for_package_score": package_score,
+        }
+        return cls.objects.create(**final_data)
```

Comment:
Why not create in directly and have defaults? Doesn't seem like we are doing much here

Reply:
the json keys are different thats why made a final object to change keys a/c to field name and then update the table